chemreg/compound/tests/test_models.py

```diff
@@ -45,6 +51,28 @@ def test_definedcompound_attr():
     assert inchikey.max_length == 29
 
 
+@pytest.mark.django_db(transaction=True)
+def test_query_structure_type():
+    """Test creation of `QueryStructureType` objects."""
+
+    # all the fields are non-nullable
+    with pytest.raises(IntegrityError):
+        QueryStructureTypeFactory(
+            name="ill-defined", label="Ill-Defined", short_description=None
+        )
+
+    # the name has to be free of reserved characters
+    qst1 = QueryStructureTypeFactory(name="ill defined", label="Ill Defined")
+    assert qst1.name == "ill-defined"
+
+    qst2 = QueryStructureTypeFactory(name="$omething weird")
+    assert qst2.name == "omething-weird"
```

Comment:
```suggestion
    model = QueryStructureTypeFactory.build(name="ill defined")
    model.clean()
    assert model.name == "ill-defined"

    model = QueryStructureTypeFactory.build(name="$omething weird")
    model.clean()
    assert model.name == "omething-weird"
```
(assuming the name is slugified in the clean method)
Example of non DB test.